examples/parent_child.py

```diff
@@ -0,0 +1,252 @@
+
+    def save(self, **kwargs):
+        # if there is no date, use now
+        if self.created is None:
+            self.created = datetime.now()
+        return super(Post, self).save(**kwargs)
+
+class Question(Post):
+    # use multi True so that .tags will return empty list if not present
+    tags = Keyword(multi=True)
+    title = Text(fields={'keyword': Keyword()})
+
+    @classmethod
+    def search(cls, **kwargs):
+        return base_search(**kwargs).filter('term', question_answer='question')
+
+    def add_answer(self, user, body, created=None, accepted=False, commit=True):
+        answer = Answer(
+            # required make sure the answer is stored in the same shard
+            _routing=self.meta.id,
+            # since we don't have explicit index, ensure same index as self
+            _index=self.meta.index,
+            # set up the parent/child mapping
+            question_answer={'name': 'answer', 'parent': self.meta.id},
+
+            # pass in the field values
+            author=user,
+            created=created,
+            body=body,
+            accepted=accepted
+        )
+        if commit:
+            answer.save()
+        return answer
+
+    def search_answers(self):
+        # search only our index
+        s = Answer.search(index=self.meta.index)
+        # filter for answers belonging to us
+        s = s.filter('parent_id', type="answer", id=self.meta.id)
+        # add routing to only go to specific shard
+        s = s.params(routing=self.meta.id)
+        return s
+
+    def save(self, **kwargs):
+        self.question_answer = 'question'
+        return super(Question, self).save(**kwargs)
+
+    class Meta:
+        def matches(hit):
+            " Use Question class for parent documents "
+            return hit['_source']['question_answer'] == 'question'
+
+
+class Answer(Post):
+    is_accepted = Boolean()
+
+    @classmethod
+    def search(cls, **kwargs):
+        return super(Answer, cls).search(**kwargs).exclude('term', question_answer='question')
+
+    @property
+    def question(self):
+        # cache question in self.meta
+        # any attributes set on self would be interpretted as fields
+        if 'question' not in self.meta:
+            self.meta.question = Question.get(
+                    id=self.question_answer.parent, index=self.meta.index)
+        return self.meta.question
+
+    def save(self, **kwargs):
+        self.meta.routing = self.question_answer.parent
+        return super(Answer, self).save(**kwargs)
+
+    class Meta:
+        def matches(hit):
+            " Use Answer class for child documents with child name 'answer' "
+            return isinstance(hit['_source']['question_answer'], dict) and hit['_source']['question_answer'].get('name') == 'answer'
+
+
+# construct basic index and populate with default settings. This will then be
+# used to create a template
+BASE_INDEX = Index('*.stackexchange.org')
```

Comment:
Creating Index with "*" fails

"elasticsearch.exceptions.RequestError: TransportError(400, 'invalid_index_name_exception', 'Invalid index name [*.stackexchange.org], must not contain the following characters [ , ", *, \\, <, |, ,, >, /, ?]')"

Reply:
This index should never be created directly. Always create an index returned by `get_index` function as the example shows